editor/editor_data.h

```diff
@@ -176,6 +197,20 @@ class EditorData {
 	bool has_extension_editor_plugin(const StringName &p_class_name);
 	EditorPlugin *get_extension_editor_plugin(const StringName &p_class_name);
 
+	// context menu plugin.
+	void add_context_menu_plugin(ContextMenuSlot slot, const Ref<EditorContextMenuPlugin> &p_plugin);
+	void remove_context_menu_plugin(ContextMenuSlot slot, const Ref<EditorContextMenuPlugin> &p_plugin);
+	//
+	void add_options_from_plugins(PopupMenu *p_popup, ContextMenuSlot p_slot, const Vector<String> &p_paths);
+	// for filesystem dock.
```

Comment:
Instead of commenting, you could rename the methods. They don't need to use the same name.

Reply:
Earlier, I thought that if we clear after adding options, the subsequent event responses wouldn't be able to find the functions. Now, I believe this clearing should happen before adding options to the plugin. The other suggestions are very useful, and the +1 for starting the ID from 1 is not essential. Thank you! 👋~